src/platform/input/game-pads.js

```diff
@@ -74,56 +115,600 @@ const MAPS = {
             'PAD_R_STICK_X',
             'PAD_R_STICK_Y'
         ]
+    },
+
+    DEFAULT_XR: {
+        buttons: [
+            // Back buttons
+            'XRPAD_TRIGGER',
+            'XRPAD_SQUEEZE',
+
+            // Axes buttons
+            'XRPAD_TOUCHPAD_BUTTON',
+            'XRPAD_STICK_BUTTON',
+
+            // Face buttons
+            'XRPAD_A',
+            'XRPAD_B'
+        ],
+
+        axes: [
+            // Analogue Sticks
+            'XRPAD_TOUCHPAD_X',
+            'XRPAD_TOUCHPAD_Y',
+            'XRPAD_STICK_X',
+            'XRPAD_STICK_Y'
+        ]
     }
 };
 
 const PRODUCT_CODES = {
     'Product: 0268': 'PS3'
 };
 
+let deadZone = 0.25;
+
+/**
+ * A GamePadButton stores information about a button from the Gamepad API.
+ */
+class GamePadButton {
+    /**
+     * Create a new GamePadButton instance.
+     *
+     * @param {GamepadButton} button - The original Gamepad API gamepad button.
+     * @hideconstructor
+     */
+    constructor(button) {
+        /**
+         * Whether this button was pressed on last frame.
+         *
+         * @type {boolean}
+         * @ignore
+         */
+        this._previouslyPressed = button.pressed;
+
+        /**
+         * Whether this button was touched on last frame.
+         *
+         * @type {boolean}
+         * @ignore
+         */
+        this._previouslyTouched = typeof button.touched === 'boolean' ? button.touched : button.value > 0;
+
+        /**
+         * The original Gamepad API gamepad button.
+         *
+         * @type {GamepadButton}
+         * @ignore
+         */
+        this._button = button;
+    }
+
+    /**
+     * Update the existing GamePadButton Instance.
+     *
+     * @param {GamepadButton} button - The original Gamepad API gamepad button.
+     * @ignore
+     */
+    update(button) {
+        this._button = button;
+    }
+
+    /**
+     * Update the previous values for this button.
+     *
+     * @ignore
+     */
+    updatePrevious() {
+        this._previouslyPressed = this.pressed;
+        this._previouslyTouched = this.touched;
+    }
+
+    /**
+     * The value for the button between 0 and 1, with 0 representing a button that is not pressed, and 1 representing a button that is fully pressed.
+     *
+     * @type {number}
+     */
+    get value() {
+        return this._button.value;
+    }
+
+    /**
+     * Whether the button is currently down.
+     *
+     * @type {boolean}
+     */
+    get pressed() {
+        return this._button.pressed;
+    }
+
+    /**
+     * Whether the button was pressed.
+     *
+     * @type {boolean}
+     */
+    get wasPressed() {
+        return this._previouslyPressed === false && this.pressed;
+    }
+
+    /**
+     * Whether the button was released since the last update.
+     *
+     * @type {boolean}
+     */
+    get wasReleased() {
+        return this._previouslyPressed === true && !this.pressed;
```

Comment:
`=== true` is just nothing since you declared the type as boolean?

Reply:
That's fixed! Thanks